src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -121,15 +121,56 @@ bool DefaultOTARequestorDriver::ProviderLocationsEqual(const ProviderLocationTyp
     }
 }
 
-void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error) {}
+void DefaultOTARequestorDriver::HandleError(UpdateFailureState state, CHIP_ERROR error)
+{
+    if (error == CHIP_NO_ERROR)
+    {
+        return;
+    }
+
+    switch (state)
+    {
+    case UpdateFailureState::kUnknown:
+        // Retry with next provider in the list
+        ScheduleRetry(false);
+        break;
+    case UpdateFailureState::kIdle:
+        // Ignore - We shouldn't be hitting this case.
+        break;
+    case UpdateFailureState::kQuerying:
+        if (error != CHIP_ERROR_BUFFER_TOO_SMALL)
+        {
+            // Retry with same provider
+            ScheduleRetry(true);
+        }
+        break;
+    case UpdateFailureState::kDownloading:
+        // Retry with same provider
+        ScheduleRetry(true);
+        break;
+    case UpdateFailureState::kApplying:
+        // Retry with same provider
+        ScheduleRetry(true);
+        break;
+    case UpdateFailureState::kNotifying:
+        // Ignore error and transition to Idle
```

Comment:
nit: Probably no need to mention `transition to idle` as this is logic in the caller (DefaultOTARequestor). Would probably also be helpful to explain why we're ignoring the error. Same for the `kAwaitingNextAction` and `kDelayedOnUserConsent` cases below.

Reply:
Updated the comments.